drivers/power/ltc7841/ltc7841.c

```diff
@@ -0,0 +1,480 @@
+	if (ret) {
+		return ret;
+	}
+	// get latest data for MFR_CONFIG
+	config = g_i2c0_read_buffer[0];
+	config &= LTC7841_NOMINAL_SLEW_RATE;
+
+	ret = ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+	return ret;
+}
+/**
+ * @brief Change to Slow Slew Rate for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_slow_slew_rate(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, g_i2c0_read_buffer);
+	if (ret) {
+		return ret;
+	}
+	// get latest data for MFR_CONFIG
+	config = g_i2c0_read_buffer[0];
+	config |= LTC7841_SLOW_SLEW_RATE;
+
+	ret = ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+	return ret;
+}
+/**
+ * @brief Change to Fast Slew Rate for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_fast_slew_rate(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, g_i2c0_read_buffer);
+	if (ret) {
+		return ret;
+	}
+	// get latest data for MFR_CONFIG
+	config = g_i2c0_read_buffer[0];
+	config |= LTC7841_FAST_SLEW_RATE;
+
+	ret = ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+	return ret;
+}
+/**
+ * @brief Change to Internal Die Temeperature Reading for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_internal_die_temperature(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, g_i2c0_read_buffer);
+	if (ret) {
+		return ret;
+	}
+	// get latest data for MFR_CONFIG
+	config = g_i2c0_read_buffer[0];
+	config &= ((LTC7841_INTERNAL_DIE_TEMPERATURE) << READ_TEMPERATURE_1_BIT);
+	ret = ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+	return ret;
+}
+/**
+ * @brief Change to TSNS PIN VOLTAGE for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_tsns_pin_voltage(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, g_i2c0_read_buffer);
+	if (ret) {
+		return ret;
+	}
+	// get latest data for MFR_CONFIG
+	config = g_i2c0_read_buffer[0];
+	config |= ((LTC7841_TSNS_PIN_VOLTAGE) << READ_TEMPERATURE_1_BIT);
+	ret = ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+	return ret;
+}
+/**
+ * @brief Change Vout command
+ * @param desc - The device structure.
+ * @param vout_level - value is from 0 to 0x1FF
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_vout_command(struct ltc7841_desc *desc, uint16_t vout_level)
+{
+	int ret;
+
+	if (vout_level <= MFR_MAX_VALUE) {
+		// okay to proceed
+		ret = ltc7841_reg_write(desc, LTC7841_MFR_VOUT_COMMAND, vout_level);
+		if (ret) {
+			return ret;
+		} else {
+			ret = -1;
+		}
+	}
+	return ret;
```

Comment:
CI checks fail here because outer if has no else therefore, ret has a chance of not getting initialized.

Reply:
I didn't write my code this way, formatting with astyle.exe caused this issue.

My original code was :
int ltc7841_change_vout_command(struct ltc7841_desc *desc, uint16_t vout_level)
{
	int ret;

	if (vout_level <= MFR_MAX_VALUE) {
		// okay to proceed
		ret = ltc7841_reg_write(desc, LTC7841_MFR_VOUT_COMMAND, vout_level);
		if (ret) 
		{
			return ret;
		} 
	}
	else
	{
		ret = -1;
	}
	return ret;
}

astyle formatting made it so that the outer else turned into an inner else for some reason

Reply:
I'm changing it now and will double check after formatting with astyle to be sure

Reply:
Also, don't put {} if there is only a single statement.

Please use the errors defined in errno.h.

Another thing, macros should be prepended with LTC7841 i.e. MFR_MAX_VALUE -> LTC7841_MFR_MAX_VALUE